pkg/rpc/writer.go

```diff
@@ -122,16 +133,54 @@ func (ow *OutputWriter) StdoutWriter() io.Writer {
 	return &stdoutWriter{ow}
 }
 
+// binaryWriter implements io.Writer, and passes all writes to the OutputWriter.WriteBinary()
+// to marshal into chunk.Binary JSON messages.
+type binaryWriter struct{ ow *OutputWriter }
+
+var _ io.Writer = (*binaryWriter)(nil)
+
+func (bw *binaryWriter) Write(p []byte) (n int, err error) {
+	return bw.ow.WriteBinary(p)
+}
+
+func (ow *OutputWriter) BinaryWriter() io.Writer {
+	return ow.bw
+}
+
 // With returns a new OutputWriter, replacing the SugaredLogger with the result
 // from delegating to SugaredLogger.With.
 func (ow *OutputWriter) With(args ...interface{}) *OutputWriter {
 	return &OutputWriter{
-		SugaredLogger:  ow.SugaredLogger.With(args...),
-		out:            ow.out,
-		progressWriter: ow.progressWriter,
+		SugaredLogger: ow.SugaredLogger.With(args...),
+		out:           ow.out,
+		pw:            ow.pw,
 	}
 }
 
+func (ow *OutputWriter) WriteProgress(b []byte) (n int, err error) {
+	return ow.pw.Write(b)
+}
+
+func (ow *OutputWriter) WriteBinary(b []byte) (n int, err error) {
+	msg := Chunk{Type: ChunkTypeBinary, Payload: base64.StdEncoding.EncodeToString(b)}
```

Comment:
No need to convert the `[]byte` to a base64 string manually. The `json` package already does this for us: https://golang.org/pkg/encoding/json/#Marshal.